axlearn/common/learner.py

```diff
@@ -94,8 +95,7 @@ def _prune_empty(in_tree: NestedTensor) -> NestedTensor:
     return prune_tree(in_tree, lambda _, v: isinstance(v, dict) and not v)
 
 
-@dataclasses.dataclass
-class ForwardOutputs:
+class ForwardOutputs(NamedTuple):
```

Comment:
What is the purpose of this change?

Reply:
The interface of jax.lax.scan does not consider "_dataclass_" a valid JAX type.